aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2141,36 +2141,56 @@ std::vector<Tensor> meshgrid(TensorList tensors,
   int64_t size = tensors.size();
   TORCH_CHECK(size > 0, "meshgrid expects a non-empty TensorList");
 
+  std::vector<std::reference_wrapper<const Tensor>> tensor_refs(tensors.begin(),
```

Comment:
Comment here explaining why this datastructure is created

Reply:
good call. done.